### PR TITLE
Update icon to be compatible with propshaft

### LINKED ISSUE #2669: Undefind method find_asset when using propshaft
https://github.com/projectblacklight/blacklight/blob/main/app/models/blacklight/icon.rb#L74

### PR DESCRIPTION
**Pull request by jcoyne**
Propshaft is a simplified asset pipeline for rails 7. See https://github.com/rails/propshaft#propshaft

Fixes #2669

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -101,7 +101,7 @@ jobs:
       run: bundle exec rake ci
       env:
         RAILS_VERSION: 7.0.0
-        ENGINE_CART_RAILS_OPTIONS: '--skip-git --skip-keeps --skip-action-cable --skip-test'
+        ENGINE_CART_RAILS_OPTIONS: '-a propshaft --skip-git --skip-keeps --skip-action-cable --skip-test'
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply .github/workflows/ruby.yml:104]: This requires https://github.com/cbeer/engine_cart/pull/112

**Comment by barmintor** [comment]
I think this isn't necessarily exclusive to #2670, so I'm inclined to take this forward after a rebase - especially since the `Icon` model change seems beneficial regardless.

**Comment by barmintor** [comment]
Rebased and running CI

**Comment by barmintor** [comment]
Also doing a better job rebasing, yeesh

**Review by barmintor — APPROVED** [review]
Looks good, let's get it moving!

**Pull request merged**